homeassistant/components/solarlog/sensor.py

```diff
@@ -38,11 +39,16 @@ def __init__(
         )
 
     @property
-    def native_value(self) -> StateType:
+    def native_value(self):
         """Return the native sensor value."""
-        result = getattr(self.coordinator.data, self.entity_description.key)
-        if self.entity_description.factor:
-            state = round(result * self.entity_description.factor, 3)
+        if self.entity_description.key == "time":
+            state = as_local(
```

Comment:
It would be nice to move this case to the entity description instead eg with a lambda function attribute.

Example:
https://github.com/home-assistant/core/blob/c69e479bfd6cc5de01f5ad809da6a05a814badcb/homeassistant/components/airly/sensor.py#L100

Reply:
Will look into it, will create a new PR when I'm finished.